### DISCUSSION TITLE
Railway部署有办法塞主题吗

### DISCUSSION THREAD

**Discussion by HinataKato** [zdz/ServerStatus-Rust#37] [Q&A] [answered]
想了下应该只能自己在Project里另外弄一个Nginx然后反向代理？

**Comment by zdz** [comment, marked as answer]
`Railway` 目前没看到自定义 `route` 的功能， 可以直接把你的主题替换 `web` 目录，然后重新编译构建部署

推荐的另一种方法是你可以在 `vercel.app` 部署前端主题，前后端分离部署，用 `routes` 指向后端，功能类似 `nginx` 反代，优点是隐藏后端和主题保持在自己的库中，主题随便修改而无需改动 `server`。目前 `ssr.rs` 也是使用这种方式。

新建一个库，然后在 `vercel` 导入部署(google `vercel静态站点部署`)，工程目录结构

```
~$ tree .
.
├── README.md
├── css
│   ├── bootstrap-theme.min.css
│   ├── bootstrap-theme.min.css.map
│   ├── bootstrap.min.css
│   ├── bootstrap.min.css.map
│   ├── dark.css
│   └── light.css
├── favicon.ico
├── google89dd963c5b122b1b.html
├── img
│   ├── dark.png
│   └── light.png
├── index.html
├── js
│   ├── bootstrap.min.js
│   ├── html5shiv.js
│   ├── jquery.min.js
│   ├── respond.min.js
│   └── serverstatus.js
├── netlify.toml
└── vercel.json
```

在  `vercel.json` 来指定其它路由，你的 `railway` 或其它方式部署的 `stat_server`
```
{
    "routes": [
        {
            "src": "/report",
            "dest": "http://tz.xxx.com:8080/report"
        },
        {
            "src": "/json/stats.json",
            "dest": "http://tz.xxx.com:8080/json/stats.json"
        },
        {
            "src": "/detail",
            "dest": "http://tz.xxx.com:8080/detail"
        },
        {
            "src": "/map",
            "dest": "http://tz.xxx.com:8080/map"
        },
        {
            "src": "/i",
            "dest": "http://tz.xxx.com:8080/i"
        }
    ]
}
```

**Reply by HinataKato** [reply]
尝试了一下确实可以 现在的问题是给原版用的Webui都会不能显示负载（因为stats.json格式不一样，原版是load Rust版本是load_数字而且我看不懂Vue）有啥解决方法吗（

**Reply by zdz** [reply]
记得我这个就是原版的。。不是vue的，哪个主题？试试查找替换吧，或者发出来等前端大佬改吧，我也不会vue，前端能力有限

**Reply by HinataKato** [reply]
> 记得我这个就是原版的。。不是vue的，哪个主题？试试查找替换吧，或者发出来等前端大佬改吧，我也不会vue，前端能力有限

https://github.com/cokemine/hotaru_theme 这个 我看了一会没看明白是怎么分析stats.json的

**Reply by HinataKato** [reply]
> > 记得我这个就是原版的。。不是vue的，哪个主题？试试查找替换吧，或者发出来等前端大佬改吧，我也不会vue，前端能力有限
> 
> https://github.com/cokemine/hotaru_theme 这个 我看了一会没看明白是怎么分析stats.json的

~~我刚才测试了一下确实就是load段不一样导致没法用~~ 好了我看起来修好了 ~~等会发一下库~~ [这里](https://github.com/HinataKato/hotaru_theme_for_RustVersion/releases/tag/v0.4.3-1) （还没完全修好）

**Reply by zdz** [reply]
赞，测试ok可以发个库链接出来，我在 README.md 可以加个主题链接

**Comment by HinataKato** [comment]
主题弄好了 这是效果图
[git库在这（点我）](https://github.com/HinataKato/hotaru_theme_for_RustVersion)
[图内版本的release（点我）](https://github.com/HinataKato/hotaru_theme_for_RustVersion/releases)

![电脑](https://user-images.githubusercontent.com/57648531/167892308-16a86df7-aa70-4617-95bf-02f6f0f851bf.png)
![手机](https://user-images.githubusercontent.com/57648531/167892318-259610cd-0fcb-4348-8674-150983a6418e.png)

**Reply by zdz** [reply]
在线时间是不是溢出了？不太对。
多嘴说一句，下面国旗那部分信息在上面不是有了么？是不是显得有些冗余？我觉得没必要为了显示国旗增加下面那部分

**Reply by HinataKato** [reply]
> 在线时间是不是溢出了？不太对。 多嘴说一句，下面国旗那部分信息在上面不是有了么？是不是显得有些冗余？我觉得没必要为了显示国旗增加下面那部分

在线时间是为了防止泄露隐私瞎改的 下面国旗部分是原版有的 你要嫌多余的话我单独开个branch删了也没问题

**Reply by zdz** [reply]
ok

**Reply by zdz** [reply]
README.md 已加上

**Comment by Tony-Yes** [comment]
想在下面国旗那里显示总流量或者本月流量应该怎么修改？js和css完全看不懂
![image](https://user-images.githubusercontent.com/72665660/207596258-ea424a64-1773-42d2-b6ce-4bebc1dbe9b3.png)
![image](https://user-images.githubusercontent.com/72665660/207596396-e9e9360e-e81b-40fb-8486-b9a7fedfa16f.png)

**Reply by zdz** [reply]
https://github.com/HinataKato/hotaru_theme_for_RustVersion/blob/4be7d40590601d54464b5bed56e81691ec62c173/src/components/CardItem.vue#L17

**Reply by Tony-Yes** [reply]
嗯，后面我看了源码知道是这里修改，但是我不会改，尴尬了
